src/dbnode/encoding/arrow/tile/series_block_iterator.go

```diff
@@ -105,7 +99,7 @@ func (b *seriesBlockIter) Next() bool {
 
 	var err error
 	for i := 0; i < b.concurrency; i++ {
-		_, b.tagIters[i], b.dataBytes[i], _, err = b.reader.Read()
+		b.ids[i], b.tagIters[i], b.dataBytes[i], _, err = b.reader.Read()
```

Comment:
On `EOF` (or error), this sets the left values to nil, which later causes panic from `Close()`.